component/pdp/capabilities.go

```diff
@@ -0,0 +1,112 @@
+package pdp
+
+import (
+	"embed"
+	"encoding/json"
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+	"github.com/zorgbijjou/golang-fhir-models/fhir-models/fhir"
+)
+
+//go:embed capabilities/*.json
+var FS embed.FS
+
+func readCapability(name string) (fhir.CapabilityStatement, error) {
+	fileName := fmt.Sprintf("capabilities/%s.json", name)
+	data, err := FS.ReadFile(fileName)
+	if err != nil {
+		return fhir.CapabilityStatement{}, err
+	}
+
+	var capability fhir.CapabilityStatement
+	if err := json.Unmarshal(data, &capability); err != nil {
+		log.Warn().Err(err).Msg(fmt.Sprintf("unable to read JSON in %s", fileName))
+		return fhir.CapabilityStatement{}, err
+	}
+
+	return capability, nil
+}
+
+func capabilityForScope(scope string) (fhir.CapabilityStatement, bool) {
+	switch scope {
+	// FUTURE: Should be made configurable or packaged up with some policy
+	case "mcsd_update":
+		capa, err := readCapability("nl-gf-admin-directory-update-client")
+		return capa, err == nil
+	case "patient_example":
+		capa, err := readCapability("patient-example")
+		return capa, err == nil
+	default:
+		return fhir.CapabilityStatement{}, false
+	}
+}
+
+func EvalCapabilityPolicy(input MainPolicyInput) PolicyResult {
+	out := PolicyResult{
+		Allow: false,
+	}
+
+	if input.Scope == "" {
+		reason := ResultReason{
+			Code:        "missing_required_value",
+			Description: "missing required value for scope field",
+		}
+		out.Reasons = []ResultReason{reason}
+		return out
+	}
+
+	statement, ok := capabilityForScope(input.Scope)
+	if !ok {
+		reason := ResultReason{
+			Code:        "unexpected_input",
+			Description: "unexpected input, no capability statement known for scope",
```

Comment:
do we always want to fail on this? Because I don't think that what was built at the hackaton, has a CapabilityStatement (and it will break if we merge this).

Ultimately, you probably want to have that configurable; which scopes are accepted, and whether (or which) capabilitystatement is authorized.